File: trunk/ServerPhp/src/dc/DC0066.php
<?php
/* N21 eBusiness Suite
 * Copyright: Nan21 Electronics srl
 * Generated content.
 * DC0066 Controller: Menu shortcuts
 */

require_once(PATH_N21LIB."/Http/Controller.php");

class DC0066 extends Controller {

  public function __construct($request, $response, $session) {
    parent::__construct($request, $response, $session);
    $this->_initFields();
  }

private function preQuery(&$params, &$where) {
    if (!empty($_REQUEST["QRY_ID"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.ID like :ID";
      $params["ID"] = $_REQUEST["QRY_ID"];
    }
    if (!empty($_REQUEST["QRY_LINK"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.LINK like :LINK";
      $params["LINK"] = $_REQUEST["QRY_LINK"];
    }
    if (!empty($_REQUEST["QRY_MENUITEM_ID"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.MENUITEM_ID like :MENUITEM_ID";
      $params["MENUITEM_ID"] = $_REQUEST["QRY_MENUITEM_ID"];
    }
    if (!empty($_REQUEST["QRY_MENUSHRCT_ID"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.MENUSHRCT_ID like :MENUSHRCT_ID";
      $params["MENUSHRCT_ID"] = $_REQUEST["QRY_MENUSHRCT_ID"];
    }
    if (!empty($_REQUEST["QRY_OWNER"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.OWNER like :OWNER";
      $params["OWNER"] = $_REQUEST["QRY_OWNER"];
    }
    if (!empty($_REQUEST["QRY_POSITION"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.POSITION like :POSITION";
      $params["POSITION"] = $_REQUEST["QRY_POSITION"];
    }
    if (!empty($_REQUEST["QRY_TITLE"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.TITLE like :TITLE";
      $params["TITLE"] = $_REQUEST["QRY_TITLE"];
    }
}

public function doQuery() {
    $start = nvl($this->getRequestParam("start"), 0);
    $limit = nvl($this->getRequestParam("limit"),20);
    $orderBy = (!empty($_REQUEST["sort"]))?$_REQUEST["sort"]:"";
    $orderSense = (!empty($_REQUEST["dir"]))?$_REQUEST["dir"]:"";
    $orderByClause = (!empty($orderBy))? "order by $orderBy $orderSense " : "" ;
    $where = "";
    $params = array();
    $this->preQuery($params, $where);
    if (!empty($where)) {
      $where = " where ".$where;
    }
    $sql = "select 
                ms.CREATEDBY
                ,ms.CREATEDON
                ,ms.ID
                ,ms.LINK
                ,ms.MENUITEM_ID
                ,ms.MENUSHRCT_ID
                ,ms.MODIFIEDBY
                ,ms.MODIFIEDON
                ,ms.OWNER
                ,ms.POSITION
                ,ms.TITLE
            from MENU_SHORTCUT ms $where $orderByClause ";
    $this->logger->debug($sql);
    $rs = $this->getDbConn()->SelectLimit($sql, $limit, $start, $params);
    $rsCount = $this->getDbConn()->Execute("select count(*) TOTALCOUNT from (".$sql.") t", $params);
    $rsCount->MoveFirst();
    $dataOut = $this->serializeCursor($rs,array_keys($this->fields), $this->getDataFormatFetch());
    if ($this->getDataFormatFetch() == "xml" ) {header("Content-type: application/xml");}
    print $this->formatQueryResponseData($dataOut, $rsCount->fields["TOTALCOUNT"]);
} /* end function doQuery  */


public function doExport() {
    $start = nvl($this->getRequestParam("start"), 0);
    $limit = nvl($this->getRequestParam("limit"),20);
    $groupBy = (!empty($_REQUEST["groupBy"]))?$_REQUEST["groupBy"]:"";
    $orderBy = (!empty($_REQUEST["sort"]))?$_REQUEST["sort"]:"";
    $orderSense = (!empty($_REQUEST["dir"]))?$_REQUEST["dir"]:"";
    $orderByClause = (!empty($orderBy))? "order by $orderBy $orderSense " : "" ;
    $where = "";
    $params = array();
    $this->preQuery($params, $where);
    if (!empty($where)) {
      $where = " where ".$where;
    }
    $sql = "select 
                ms.ID
                ,ms.POSITION
                ,ms.OWNER
                ,ms.TITLE
                ,ms.LINK
                ,ms.MENUSHRCT_ID
                ,ms.MENUITEM_ID
                ,ms.CREATEDON
                ,ms.CREATEDBY
                ,ms.MODIFIEDON
                ,ms.MODIFIEDBY
            from MENU_SHORTCUT ms $where $orderByClause ";
    $rs = $this->getDbConn()->Execute($sql, $params);
    $rsCount = $this->getDbConn()->Execute("select count(*) TOTALCOUNT from (".$sql.") t", $params);
    $rsCount->MoveFirst();
    if (!empty($_REQUEST["_p_disp_cols"])) {
      $columns = explode("|",$_REQUEST["_p_disp_cols"]);
    } else {
      $columns = array_keys($this->fields);
    }
    if ($this->getDataFormatExport() == "csv" ) {
      $dataOut = $this->serializeCursor($rs,$columns,"csv");
    } else {
      $dataOut = $this->serializeCursor($rs,$columns,"xml");
      $dataOut = "<records>".$dataOut."</records>";
      $dataOut = "<queryParams>".$this->serializeMap($params,"xml")."</queryParams>".$dataOut;
      $dataOut = "<columnDef>".$this->columnDefForExport($columns,$this->fields,true).$this->columnDefForExport(array_diff(array_keys($params), $columns),$this->fields,false)."</columnDef>".$dataOut;
      $dataOut = "<staticText>".$this->exportLocalizedStaticText()."</staticText>".$dataOut;
      $dataOut = "<groupBy>".$groupBy."</groupBy>".$dataOut;
      $dataOut = "<reportData  title=\"".$this->getDcTitle()."\" by=\"".$_SESSION["user"]["userName"]."\" on=\"".date(DATE_FORMAT)."\">".$dataOut."</reportData>";
    }
    $this->beginExport();
    print $dataOut;
    $this->endExport();
} /* end function doExport  */

public function fetchRecord() {
    $this->record["ID"] = $this->getRequestParam("ID");
    if (empty($this->record["ID"])) { throw new Exception("Missing value for primary key field ID in DC0066.fetchRecord().");}
    $pkArray = array("ID" => $this->record["ID"]);
    $this->findByPk($pkArray, $this->record);
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function fetchRecord */

public function doInsert() {
    $this->populateRecordFromRequest(); 
    $this->record["_p_record_status"] = $this->getRequestParam("_p_record_status");
    $this->record["_p_store_recId"] = $this->getRequestParam("_p_store_recId");
    $sql = "insert into MENU_SHORTCUT(
                 ID
                ,LINK
                ,MENUITEM_ID
                ,MENUSHRCT_ID
                ,OWNER
                ,POSITION
                ,TITLE
            ) values ( 
                 :ID
                ,:LINK
                ,:MENUITEM_ID
                ,:MENUSHRCT_ID
                ,:OWNER
                ,:POSITION
                ,:TITLE
    )";
    $stmt = $this->getDbConn()->prepare($sql);
    $_seq = $this->getDbConn()->execute("select SEQ_MENUSHRCT_ID.nextval seq_val from dual")->fetchRow();
    $this->record["ID"] = $_seq["SEQ_VAL"];
    $this->getDbConn()->Execute($stmt, $this->record);
    $pkArray = array("ID" => $this->record["ID"]);
    $this->findByPk($pkArray, $this->record);
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function doInsert */

public function doUpdate() {
    $this->populateRecordFromRequest();
    $this->record["_p_record_status"] = $this->getRequestParam("_p_record_status");
    $this->record["_p_store_recId"] = $this->getRequestParam("_p_store_recId");
    $sql = "update MENU_SHORTCUT set 
                 CREATEDBY=:CREATEDBY
                ,CREATEDON=:CREATEDON
                ,LINK=:LINK
                ,MENUITEM_ID=:MENUITEM_ID
                ,MENUSHRCT_ID=:MENUSHRCT_ID
                ,MODIFIEDBY=:MODIFIEDBY
                ,MODIFIEDON=:MODIFIEDON
                ,OWNER=:OWNER
                ,POSITION=:POSITION
                ,TITLE=:TITLE
    where 
           ID= :ID
    ";
    $stmt = $this->getDbConn()->prepare($sql);
    $this->getDbConn()->Execute($stmt, $this->record);
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function doUpdate */

public function doDelete() {
    $this->record["ID"] = $this->getRequestParam("ID");
    if (empty($this->record["ID"])) { throw new Exception("Missing value for primary key field ID in DC0066.doDelete().");}
    $sql = "delete from MENU_SHORTCUT where 
           ID= :ID
    ";
    $stmt = $this->getDbConn()->prepare($sql);
    $this->getDbConn()->Execute($stmt, $this->record);
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function doDelete */

public function initNewRecord() {
    $this->populateRecordFromRequest();
    $this->record["_p_record_status"] = "insert";
    $this->setFieldInitialValues($this->record, "DC0066");
    print "{success:true, data:".json_encode($this->record)."}";
} /* end function initNewRecord */

private function findByPk(&$pkCols, &$record) {
    $sql = "select 
                ms.CREATEDBY
                ,ms.CREATEDON
                ,ms.ID
                ,ms.LINK
                ,ms.MENUITEM_ID
                ,ms.MENUSHRCT_ID
                ,ms.MODIFIEDBY
                ,ms.MODIFIEDON
                ,ms.OWNER
                ,ms.POSITION
                ,ms.TITLE
            from MENU_SHORTCUT ms
         where 
           ms.ID= :ID
            ";
    $rs = $this->getDbConn()->Execute($sql, $pkCols);
    $row = $rs->FetchRow();
    CollectionUtils::copyMapValues($record, $row);
} /* end function findByPk  */


public function callProcedure($pName) {
  $this->logger->debug("Start: ".$this->dcName.".callProcedure(".$pName.")");
    $this->populateRecordFromRequest();
    print "{success:true, data:".json_encode($this->record)."}";
    $this->logger->debug("End: ".$this->dcName.".callProcedure");
} /* end function callProcedure */


private function _initFields() {
  $this->fields = array(
  "CREATEDBY" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"CREATEDON" => array(parent::FLDPROP_DATA_TYPE => "DATE")
  ,"ID" => array(parent::FLDPROP_DATA_TYPE => "NUMBER")
  ,"LINK" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"MENUITEM_ID" => array(parent::FLDPROP_DATA_TYPE => "NUMBER")
  ,"MENUSHRCT_ID" => array(parent::FLDPROP_DATA_TYPE => "NUMBER")
  ,"MODIFIEDBY" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"MODIFIEDON" => array(parent::FLDPROP_DATA_TYPE => "DATE")
  ,"OWNER" => array(parent::FLDPROP_DATA_TYPE => "STRING")
  ,"POSITION" => array(parent::FLDPROP_DATA_TYPE => "NUMBER")
  ,"TITLE" => array(parent::FLDPROP_DATA_TYPE => "STRING")
);
}

}
?>

File: ServerJava/src/net/nan21/ebs/dc/DC0066.java
/* N21 eBusiness Suite
 * Copyright: Nan21 Electronics srl
 * Generated content.
 * DC0066 DC Controller: Menu shortcuts
 */

package net.nan21.ebs.dc;


import java.util.*;
import javax.servlet.http.HttpServletResponse;
import net.nan21.lib.*;

public class DC0066 extends AbstractDataControl implements IDataControl {

  public void init (HttpRequest request, HttpServletResponse response, HttpSession session, DbManager dbm) throws Exception {
    this._initFields();
    super.init(request, response, session, dbm);
  }

private void preQuery() {
    if (this.request.getParam("QRY_ID") != null && !this.request.getParam("QRY_ID").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.ID like :ID");
      this.queryParams.put("ID",(String)this.request.getParam("QRY_ID"));
    }
    if (this.request.getParam("QRY_OWNER") != null && !this.request.getParam("QRY_OWNER").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.OWNER like :OWNER");
      this.queryParams.put("OWNER",(String)this.request.getParam("QRY_OWNER"));
    }
    if (this.request.getParam("QRY_TITLE") != null && !this.request.getParam("QRY_TITLE").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.TITLE like :TITLE");
      this.queryParams.put("TITLE",(String)this.request.getParam("QRY_TITLE"));
    }
    if (this.request.getParam("QRY_LINK") != null && !this.request.getParam("QRY_LINK").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.LINK like :LINK");
      this.queryParams.put("LINK",(String)this.request.getParam("QRY_LINK"));
    }
    if (this.request.getParam("QRY_MENUITEM_ID") != null && !this.request.getParam("QRY_MENUITEM_ID").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.MENUITEM_ID like :MENUITEM_ID");
      this.queryParams.put("MENUITEM_ID",(String)this.request.getParam("QRY_MENUITEM_ID"));
    }
    if (this.request.getParam("QRY_MENUSHRCT_ID") != null && !this.request.getParam("QRY_MENUSHRCT_ID").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.MENUSHRCT_ID like :MENUSHRCT_ID");
      this.queryParams.put("MENUSHRCT_ID",(String)this.request.getParam("QRY_MENUSHRCT_ID"));
    }
    if (this.request.getParam("QRY_POSITION") != null && !this.request.getParam("QRY_POSITION").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.POSITION like :POSITION");
      this.queryParams.put("POSITION",(String)this.request.getParam("QRY_POSITION"));
    }
}

public void doQuery() throws Exception {
    this.prepareQueryContext();
    this.preQuery();
    this.queryWhere.insert(0, (this.queryWhere.length()>0)?" where ":"");
    String sql = "select "+ 
               " ms.ID"+
               " ,ms.OWNER"+
               " ,ms.TITLE"+
               " ,ms.LINK"+
               " ,ms.MENUITEM_ID"+
               " ,ms.MENUSHRCT_ID"+
               " ,ms.POSITION"+
               " ,ms.CREATEDON"+
               " ,ms.CREATEDBY"+
               " ,ms.MODIFIEDON"+
               " ,ms.MODIFIEDBY"+
           " from MENU_SHORTCUT ms "+this.queryWhere.toString()+" "+this.queryOrderBy;
    this.writeResultDoQuery(sql);
} 


public void doExport() throws Exception {
    this.prepareQueryContext();
    this.preQuery();
    this.queryWhere.insert(0, (this.queryWhere.length()>0)?" where ":"");
    String sql = "select "+ 
               " ms.ID"+
               " ,ms.POSITION"+
               " ,ms.OWNER"+
               " ,ms.TITLE"+
               " ,ms.LINK"+
               " ,ms.MENUSHRCT_ID"+
               " ,ms.MENUITEM_ID"+
               " ,ms.CREATEDON"+
               " ,ms.CREATEDBY"+
               " ,ms.MODIFIEDON"+
               " ,ms.MODIFIEDBY"+
           " from MENU_SHORTCUT ms "+this.queryWhere.toString()+" "+this.queryOrderBy;
    this.writeResultDoExport(sql);
}

 	public void fetchRecord()  throws Exception  {
     this.populateRecordPkFromRequest();
     this.findByPk();
       this.writeResultFetchRecord();	 
   }

public void doInsert()  throws Exception {
  this.populateRecordFromRequest(); 
  this.populateRecordWithClientSpecific();
    String sql = "insert into MENU_SHORTCUT("+
               "  ID"+
               " ,OWNER"+
               " ,TITLE"+
               " ,LINK"+
               " ,MENUITEM_ID"+
               " ,MENUSHRCT_ID"+
               " ,POSITION"+
           " ) values ( "+
               "  :ID"+
               " ,:OWNER"+
               " ,:TITLE"+
               " ,:LINK"+
               " ,:MENUITEM_ID"+
               " ,:MENUSHRCT_ID"+
               " ,:POSITION"+
    ")";
    this.record.put("ID",   dbm.getSequenceNextValue("SEQ_MENUSHRCT_ID")  );
    dbm.executeStatement(sql, this.record);
    this.populateRecordPkFromRecord();
    this.findByPk();
    this.writeResultDoInsert();
}

public void doUpdate()  throws Exception {
    this.populateRecordFromRequest();
    this.populateRecordWithClientSpecific();
    String sql = "update MENU_SHORTCUT set "+
               "  CREATEDBY=:CREATEDBY"+ 
               " ,CREATEDON=:CREATEDON"+ 
               " ,LINK=:LINK"+ 
               " ,MENUITEM_ID=:MENUITEM_ID"+ 
               " ,MENUSHRCT_ID=:MENUSHRCT_ID"+ 
               " ,MODIFIEDBY=:MODIFIEDBY"+ 
               " ,MODIFIEDON=:MODIFIEDON"+ 
               " ,OWNER=:OWNER"+ 
               " ,POSITION=:POSITION"+ 
               " ,TITLE=:TITLE"+ 
   " where "+ 
     "      ID= :ID"+
    "";
    dbm.executeStatement(sql, this.record);
    this.populateRecordPkFromRecord();
    this.findByPk();
    this.writeResultDoUpdate();	
}

public void doDelete() throws Exception {
    this.populateRecordPkFromRequest();
   String sql = "delete from MENU_SHORTCUT where "+
     "      ID= :ID"+
    "";
    dbm.executeStatement(sql, this.recordPk);
   this.writeResultDoDelete();
}

public void initNewRecord() throws Exception {
    this.populateRecordFromRequest();
    this.record.put("_p_record_status", "insert");
    this.writeResultInitNewRecord();
}

private void findByPk()  throws Exception {
    String sql = "select "+ 
               " ms.ID"+
               " ,ms.OWNER"+
               " ,ms.TITLE"+
               " ,ms.LINK"+
               " ,ms.MENUITEM_ID"+
               " ,ms.MENUSHRCT_ID"+
               " ,ms.POSITION"+
               " ,ms.CREATEDON"+
               " ,ms.CREATEDBY"+
               " ,ms.MODIFIEDON"+
               " ,ms.MODIFIEDBY"+
           " from MENU_SHORTCUT ms"+
        " where "+
     "      ms.ID= :ID"+ 
          "";
    this.record = dbm.executeQuery(sql, this.recordPk).get(0); 
} 


public void doCustomAction(String pName)  throws Exception {
    this.populateRecordFromRequest();
}


	private void  _initFields() {
	  this.fields = new HashMap<String, FieldDef>();
	  this.fields.put("ID", new FieldDef("NUMBER"));
	  this.fields.put("OWNER", new FieldDef("STRING"));
	  this.fields.put("TITLE", new FieldDef("STRING"));
	  this.fields.put("LINK", new FieldDef("STRING"));
	  this.fields.put("MENUITEM_ID", new FieldDef("NUMBER"));
	  this.fields.put("MENUSHRCT_ID", new FieldDef("NUMBER"));
	  this.fields.put("POSITION", new FieldDef("NUMBER"));
	  this.fields.put("CREATEDON", new FieldDef("DATE"));
	  this.fields.put("CREATEDBY", new FieldDef("STRING"));
	  this.fields.put("MODIFIEDON", new FieldDef("DATE"));
	  this.fields.put("MODIFIEDBY", new FieldDef("STRING"));
	  String[] _pkFields = {"ID"};
	  this.pkFields = _pkFields;
	  String[] _summaryFields = {};
	  this.summaryFields = _summaryFields;
	  this.queryResultSize = 20;
	}

}
